application/controllers/EventRuleController.php

```diff
@@ -291,31 +280,49 @@ public function editAction(): void
     {
         /** @var string $ruleId */
         $ruleId = $this->params->getRequired('id');
-        $config = $this->sessionNamespace->get($ruleId);
-        if ($config === null) {
-            if ($ruleId === '-1') {
-                $config = ['id' => $ruleId];
-            } else {
-                $config = $this->fromDb((int) $ruleId);
-            }
+        $db = Database::get();
+        if ($ruleId === '-1') {
+            $config = ['id' => $ruleId];
+        } else {
+            // Casting to array is required as Connection::fetchOne actually returns stdClass and not array
+            $config = (array) $db->fetchOne(
+                Rule::on($db)->withoutColumns('timeperiod_id')->filter(Filter::equal('id', $ruleId))->assembleSelect()
+            );
         }
 
         $eventRuleForm = (new EventRuleForm())
             ->populate($config)
             ->setAction(Url::fromRequest()->getAbsoluteUrl())
-            ->on(Form::ON_SUCCESS, function ($form) use ($ruleId, $config) {
-                $config['name'] = $form->getValue('name');
-                $config['is_active'] = $form->getValue('is_active');
+            ->on(Form::ON_SUCCESS, function ($form) use ($ruleId, $db) {
                 if ($ruleId === '-1') {
-                    $redirectUrl = Url::fromPath('notifications/event-rules/add', ['id' => '-1']);
-                } else {
-                    $redirectUrl = Url::fromPath('notifications/event-rule', ['id' => $ruleId]);
+                    $db->insert('rule', [
+                        'name'          => $form->getValue('name'),
+                        'timeperiod_id' => null,
+                        'object_filter' => null,
```

Comment:
Why are you inserting null? These columns are already nullable.

Reply:
You can call `$form->getValues()` to get both values.
```suggestion
    $db->insert('rule', $this->getValues());
```